File: src/Lumineux.Application/Abstractions/Permissions.cs
namespace Lumineux.Application.Abstractions;

/// <summary>Droits applicatifs (portés par les membres du bureau).</summary>
public static class Permissions
{
    /// <summary>Droit de gérer les présences (démarrer/clôturer une session, ajouter/retirer une présence).</summary>
    public const string ManageAttendance = "manage_attendance";

    /// <summary>Droit de gérer les membres (créer, corriger, consulter les fiches).</summary>
    public const string ManageMembers = "manage_members";

    /// <summary>Droit de gérer les profils du bureau (feature 004) — écritures et administration.</summary>
    public const string ManageBureauProfiles = "manage_bureau_profiles";

    /// <summary>Droit de gérer les référentiels (feature 016) — gestion des antennes (et à terme autres nomenclatures).</summary>
    public const string ManageReferentials = "manage_referentials";
}

File: mobile/lib/features/auth/application/permissions.dart
/// Libellés FR des droits de gestion (codes renvoyés par `GET /auth/me`).
const Map<String, String> _permissionLabels = {
  'manage_attendance': 'Gérer les présences',
  'manage_members': 'Gérer les membres',
  'manage_bureau_profiles': 'Gérer les profils du bureau',
  'manage_referentials': 'Gérer les référentiels',
};

/// Vrai si l'utilisateur possède au moins un droit de gestion (rôle « Bureau »).
bool hasManagementRights(List<String> permissions) =>
    permissions.any((p) => p.startsWith('manage_'));

/// Rôle effectif dérivé des droits (jamais un rôle figé — cf. handoff design).
String roleLabel(List<String> permissions) =>
    hasManagementRights(permissions) ? 'Bureau' : 'Membre';

/// Traduit les codes de droits en libellés FR affichables.
List<String> permissionLabels(List<String> permissions) => permissions
    .map((p) => _permissionLabels[p] ?? p)
    .toList(growable: false);
